core/new-gui/src/app/workspace/types/autocomplete.interface.ts

```diff
@@ -0,0 +1,54 @@
+import { JSONSchema4 } from 'json-schema';
+
+/**
+ * The type decalaration of the response sent by **backend** when
+ * asking for source table names.
+ */
+export interface SourceTableNamesAPIResponse extends Readonly < {
+  code: number,
+  message: string
+} > { }
+
+export interface SourceTableDetails extends Readonly <{
+  tableName: string,
+  schema: SourceTableSchema
+}> { }
+
+export interface SourceTableSchema extends Readonly<{
+  attributes: ReadonlyArray<SourceTableAttribute>
+}> { }
+
+export interface SourceTableAttribute extends Readonly <{
+  attributeName: string,
+  attributeType: string
+}> { }
+
+/**
+ * The backend interface of the return object of a successful execution
+ * of autocomplete API
+ */
+export interface SuccessExecutionResult extends Readonly<{
```

Comment:
Please use different names for the type. It's the same as the executing workflow's type `SuccessExecutionResult`. Just call it `AutocompleteResult` and `AutocompleteError`. `execution` is really intended to mean executing (running) the workflow, not calling a backend API